src/SDKs/Azure.Base/data-plane/Azure.Base.Tests/EventSourceTests.cs

```diff
@@ -108,5 +106,76 @@ public async Task SendingRequestProducesEvents()
+
+                var response = await pipeline.SendRequestAsync(request, CancellationToken.None);
+
+
+                Assert.AreEqual(3, await response.ContentStream.ReadAsync(buffer, 5, 3));
+                Assert.AreEqual(2, await response.ContentStream.ReadAsync(buffer, 8, 2));
+                Assert.AreEqual(0, await response.ContentStream.ReadAsync(buffer, 0, 5));
+            }
+
+            EventWrittenEventArgs[] contentEvents = _listener.EventsById(11).ToArray();
+
+            Assert.AreEqual(2, contentEvents.Length);
+
+            Assert.AreEqual(EventLevel.Verbose, contentEvents[0].Level);
+            Assert.AreEqual("ResponseContentBlock",  contentEvents[0].EventName);
+            Assert.AreEqual(requestId,  contentEvents[0].GetProperty<string>("requestId"));
+            Assert.AreEqual(0, contentEvents[0].GetProperty<int>("blockNumber"));
+            CollectionAssert.AreEqual(new byte[] { 6, 7, 8 }, contentEvents[0].GetProperty<byte[]>("content"));
+
+            Assert.AreEqual(EventLevel.Verbose, contentEvents[1].Level);
+            Assert.AreEqual("ResponseContentBlock",  contentEvents[1].EventName);
+            Assert.AreEqual(requestId,  contentEvents[1].GetProperty<string>("requestId"));
+            Assert.AreEqual(1, contentEvents[1].GetProperty<int>("blockNumber"));
+            CollectionAssert.AreEqual(new byte[] { 9, 0 }, contentEvents[1].GetProperty<byte[]>("content"));
+
+            EventWrittenEventArgs[] errorContentEvents = _listener.EventsById(12).ToArray();
+
+            Assert.AreEqual(2, errorContentEvents.Length);
+
+            Assert.AreEqual(EventLevel.Informational, errorContentEvents[0].Level);
+            Assert.AreEqual("ErrorResponseContentBlock",  errorContentEvents[0].EventName);
+            Assert.AreEqual(requestId,  errorContentEvents[0].GetProperty<string>("requestId"));
+            Assert.AreEqual(0, errorContentEvents[0].GetProperty<int>("blockNumber"));
+            CollectionAssert.AreEqual(new byte[] { 6, 7, 8 }, errorContentEvents[0].GetProperty<byte[]>("content"));
+
+            Assert.AreEqual(EventLevel.Informational, errorContentEvents[1].Level);
+            Assert.AreEqual("ErrorResponseContentBlock",  errorContentEvents[1].EventName);
+            Assert.AreEqual(requestId,  errorContentEvents[1].GetProperty<string>("requestId"));
+            Assert.AreEqual(1, errorContentEvents[1].GetProperty<int>("blockNumber"));
+            CollectionAssert.AreEqual(new byte[] { 9, 0 }, errorContentEvents[1].GetProperty<byte[]>("content"));
+
+            // No ResponseContent and ErrorResponseContent events
+            CollectionAssert.IsEmpty(_listener.EventsById(6));
+            CollectionAssert.IsEmpty(_listener.EventsById(9));
+        }
+
+        private class NonSeekableMemoryStream: MemoryStream
+        {
+            public NonSeekableMemoryStream(byte[] buffer): base(buffer)
+            {
+            }
+
+            public override bool CanSeek => false;
```

Comment:
Should we consider overriding the `Seek` method to throw a `NotImplementedException` as well?  From what I see in the [docs](https://docs.microsoft.com/en-us/dotnet/api/system.io.memorystream.seek?view=netcore-2.2) and [reference source](https://referencesource.microsoft.com/#mscorlib/system/io/memorystream.cs) it doesn't appear that the property is checked in the method.

Reply:
Done